library/alloc/src/collections/btree/map.rs

```diff
@@ -138,8 +138,9 @@ pub struct BTreeMap<K, V> {
 #[stable(feature = "btree_drop", since = "1.7.0")]
 unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for BTreeMap<K, V> {
     fn drop(&mut self) {
-        unsafe {
-            drop(ptr::read(self).into_iter());
+        let mut me = ManuallyDrop::new(self);
```

Comment:
What is the point of a `ManuallyDrop<&mut Self>`?

Reply:
I don't know, it's what the `into_iter` invoked before did.

Reply:
I think I understand the absolute futility of `ManuallyDrop` on a reference. Though one may ask "what is the point of `ManuallyDrop` in the consuming overload of `into_iter`, and I think the answer is: it was probably necessary before Mark abolished the shared root, but now it is also silly, to a lesser degree.